app_flutter/lib/sign_in_button.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'service/google_authentication.dart';
+
+/// Widget for displaying sign in information for the current user.
+///
+/// If logged in, it will display the user's avatar. Clicking it opens a dropdown for logging out.
+/// Otherwise, a sign in button will show.
+class SignInButton extends StatelessWidget {
+  const SignInButton({@required this.authService, Key key}) : super(key: key);
+
+  final GoogleSignInService authService;
+
+  @override
+  Widget build(BuildContext context) {
+    if (authService.isAuthenticated) {
+      return PopupMenuButton<String>(
+        child: Image.network(authService.user.photoUrl),
```

Comment:
There's a `GoogleUserCircleAvatar` widget provided by the `google_sign_in` plugin that you may be able to use. It's used in the [example](https://github.com/flutter/plugins/blob/master/packages/google_sign_in/google_sign_in/example/lib/main.dart#L115-L117) app.

Reply:
Using `GoogleUserCircleAvatar` would throw 2 exceptions in the tests, but I could only catch the last one with `WidgetTester.takeException()` and it would fail because the first was not caught. So for now, I am just going to leave a TODO :)